rotkehlchen/db/snapshots.py

```diff
@@ -0,0 +1,108 @@
+import csv
+import tempfile
+from pathlib import Path
+from typing import Any, Dict, List, Tuple
+from zipfile import ZipFile
+
+from rotkehlchen.accounting.export.csv import CSVWriteError
+from rotkehlchen.accounting.structures.balance import BalanceType
+from rotkehlchen.db.dbhandler import DBHandler
+from rotkehlchen.db.utils import DBAssetBalance, LocationData
+from rotkehlchen.types import Location, Timestamp
+
+
+class Snapshot:
+    def __init__(self, db_handler: DBHandler) -> None:
+        self.db_handler = db_handler
+        self.tmp_dir = tempfile.TemporaryDirectory()
+        self.tmp_dir_path = Path(self.tmp_dir.name)
+        self.snapshot_archive_path = self.tmp_dir_path / 'snapshot.zip'
+        self.timed_balances_csv_path = self.tmp_dir_path / 'timed_balances.csv'
+        self.timed_location_data_csv_path = self.tmp_dir_path / 'timed_location_data.csv'
+
+    def get_timed_balances(self, timestamp: Timestamp) -> List[Dict[str, Any]]:
+        balances_data = []
+        cursor = self.db_handler.conn.cursor()
+        timed_balances_result = cursor.execute(
+            'SELECT category, time, amount, currency, usd_value FROM timed_balances '
+            'WHERE time=?', (timestamp,),
+        )
+        for data in timed_balances_result:
+            balances_data.append(
+                DBAssetBalance(
+                    category=BalanceType.deserialize_from_db(data[0]),
+                    time=data[1],
+                    amount=data[2],
+                    asset=data[3],
+                    usd_value=data[4],
+                ),
+            )
+        return self.clean_data(balances_data)
```

Comment:
Don't return cleaned data from here. Do it only at the api exit level. This should return a list of `DBAssetBalance()`